src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -433,23 +453,49 @@ private void PopulateReferenceIdDictionary(IEnumerable<EntryComponent> bundleEnt
         {
             foreach (EntryComponent entry in bundleEntries)
             {
-                if (entry.Request.Method != HTTPVerb.POST)
-                {
-                    continue;
-                }
+                var versionId = string.Empty;
 
-                // We've already come across this ID
-                if (!string.IsNullOrWhiteSpace(entry.FullUrl) && !idDictionary.ContainsKey(entry.FullUrl))
+                if (!string.IsNullOrWhiteSpace(entry.FullUrl))
                 {
-                    // This id is new to us
-                    var insertId = _resourceIdProvider.Create();
-                    entry.Resource.Id = insertId;
+                    if (entry.FullUrl.Contains("/_history/", StringComparison.OrdinalIgnoreCase))
+                    {
+                        string[] versionedReference = entry.FullUrl.Split("/_history/");
+                        versionId = versionedReference[1];
```

Comment:
`versionId` could be replaced with a `bool` since we don't care about the actual value. It may be able to be simplified as something like `bool hasVersion = entry.FullUrl.Contains("/_history/", StringComparison.OrdinalIgnoreCase);`

Reply:
> versionId could be replaced with a bool since we don't care about the actual value. It may be able to be simplified as something like bool hasVersion = entry.FullUrl.Contains("/_history/", StringComparison.OrdinalIgnoreCase);

I have already made this change. This is old checkin. The latest change onky checks if _history is present or not in a reference.